File: akka-http-core/src/main/scala/akka/http/model/HttpProtocol.scala
/**
 * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
 */

package akka.http.model

import akka.http.util.{ SingletonValueRenderable, ObjectRegistry }

/** The protocol of an HTTP message */
final case class HttpProtocol private[http] (override val value: String) extends japi.HttpProtocol with SingletonValueRenderable

object HttpProtocols extends ObjectRegistry[String, HttpProtocol] {
  private def register(p: HttpProtocol): HttpProtocol = register(p.value, p)

  val `HTTP/1.0` = register(HttpProtocol("HTTP/1.0"))
  val `HTTP/1.1` = register(HttpProtocol("HTTP/1.1"))
}

File: akka-http-core/src/main/java/akka/http/model/japi/HttpProtocol.java
/**
 * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
 */

package akka.http.model.japi;

/**
 * Represents an Http protocol (currently only HTTP/1.0 or HTTP/1.1). See {@link HttpProtocols}
 * for the predefined constants for the supported protocols.
 */
public abstract class HttpProtocol {
    /**
     * Returns the String representation of this protocol.
     */
    public abstract String value();
}
